File: collect_app/src/main/java/org/odk/collect/android/instancemanagement/FinalizeAllSnackbarPresenter.java
package org.odk.collect.android.instancemanagement;

import android.content.Context;
import android.view.View;

import androidx.annotation.NonNull;

import org.odk.collect.android.formmanagement.FinalizeAllResult;
import org.odk.collect.androidshared.ui.SnackbarUtils;

public class FinalizeAllSnackbarPresenter extends SnackbarUtils.SnackbarPresenterObserver<FinalizeAllResult> {
    private final Context context;

    public FinalizeAllSnackbarPresenter(@NonNull View parentView, Context context) {
        super(parentView);
        this.context = context;
    }

    @NonNull
    @Override
    public SnackbarUtils.SnackbarDetails getSnackbarDetails(FinalizeAllResult result) {
        if (result.getUnsupportedInstances()) {
            return new SnackbarUtils.SnackbarDetails(
                    context.getString(
                            org.odk.collect.strings.R.string.bulk_finalize_unsupported,
                            result.getSuccessCount()
                    )
            );
        } else if (result.getFailureCount() == 0) {
            return new SnackbarUtils.SnackbarDetails(
                    context.getResources().getQuantityString(
                            org.odk.collect.strings.R.plurals.bulk_finalize_success,
                            result.getSuccessCount(),
                            result.getSuccessCount()
                    )
            );
        } else if (result.getSuccessCount() == 0) {
            return new SnackbarUtils.SnackbarDetails(
                    context.getResources().getQuantityString(
                            org.odk.collect.strings.R.plurals.bulk_finalize_failure,
                            result.getFailureCount(),
                            result.getFailureCount()
                    )
            );
        } else {
            return new SnackbarUtils.SnackbarDetails(
                    context.getString(org.odk.collect.strings.R.string.bulk_finalize_partial_success, result.getSuccessCount(), result.getFailureCount())
            );
        }
    }
}

File: collect_app/src/main/java/org/odk/collect/android/instancemanagement/FinalizeAllSnackbarPresenter.kt
package org.odk.collect.android.instancemanagement

import android.content.Context
import android.view.View
import org.odk.collect.androidshared.ui.SnackbarUtils.SnackbarDetails
import org.odk.collect.androidshared.ui.SnackbarUtils.SnackbarPresenterObserver
import org.odk.collect.strings.R

class FinalizeAllSnackbarPresenter(parentView: View, private val context: Context) :
    SnackbarPresenterObserver<FinalizeAllResult>(parentView) {

    override fun getSnackbarDetails(value: FinalizeAllResult): SnackbarDetails {
        return if (value.unsupportedInstances) {
            SnackbarDetails(
                context.getString(
                    R.string.bulk_finalize_unsupported,
                    value.successCount
                )
            )
        } else if (value.failureCount == 0) {
            SnackbarDetails(
                context.resources.getQuantityString(
                    R.plurals.bulk_finalize_success,
                    value.successCount,
                    value.successCount
                )
            )
        } else if (value.successCount == 0) {
            SnackbarDetails(
                context.resources.getQuantityString(
                    R.plurals.bulk_finalize_failure,
                    value.failureCount,
                    value.failureCount
                )
            )
        } else {
            SnackbarDetails(
                context.getString(
                    R.string.bulk_finalize_partial_success,
                    value.successCount,
                    value.failureCount
                )
            )
        }
    }
}
